Csar Multiple Changes

marcos2221 opened the pull request:
Added some small changes to have the option to generate a CSAR in case of crash or Death.
This is great for helicopter crashes etc

marcos2221 committed:
Update CSAR.lua

Generates Csar every time even when crash

marcos2221 committed:
Update CSAR.lua

Csar on crash,.  added config on top to activate / deactivate

marcos2221 committed:
Update README.md

csar.csarOncrash Update documentation

marcos2221 committed:
Update CSAR.lua

marcos2221 committed:
Update CSAR.lua

1) Csar Selection : Change from the unit name deffinition to two options csar.csarunits : should't be changed anymore
a)EnableAll Slots = will enable all helicopters (defined in csar.AircraftType) to act as csar
b)Prefix : WIll add units which name contains one of the prefix in the list.

2) Aircraft Type / Limit: Allows to set up a limit per Aircraft

3) csar.autosmoke = Activates or deactivates the automatic smoke

4) csar.allowDownedPilotCAcontrol, allows the player to control the Downed pilot with Combined arms

5) csar weight.  Simulates Downed pilot weight increase / decrease

6) Add option to the radio menu to request smoke signal (Must be inside 8km)

marcos2221 committed:
Update README.md

marcos2221 commented:
1) Csar Selection : Added to two options besides units names to set up the csar aircrafts
a)EnableAll Slots = will enable all helicopters (defined in csar.AircraftType) to act as csar (helicopters only)
b)Prefix : WIll add units which name contains one of the prefix in the list. (helicopters only)

2) Aircraft Type / Limit: Allows to set up a limit per Aircraft

3) csar.autosmoke = Activates or deactivates the automatic smoke

4) csar.allowDownedPilotCAcontrol, allows the player to control the Downed pilot with Combined arms

5) csar weight.  Simulates Downed pilot weight increase / decrease

6) Add option to the radio menu to request smoke signal (Must be inside 8km)

7) Option to enable CSAR on crash or death..

8) Pilot Run to rescue helicopter

9) Added function to create csar in trigger zone.

marcos2221 committed:
Update CSAR.lua

Added option to have the Pilot to run to the Rescue helicopter.

marcos2221 committed:
Update README.md

ciribob commented on CSAR.lua:
```diff
@@ -8,99 +8,29 @@
 csar = {}
 
 -- SETTINGS FOR MISSION DESIGNER vvvvvvvvvvvvvvvvvv
-csar.csarUnits = {
```
I'd still leave this - as a lot of people name there units still like this - along with your new settings

marcos2221 replied:
Added it back and tested it..

ciribob commented on CSAR.lua:
```diff
@@ -306,7 +249,7 @@ function csar.eventHandler:onEvent(_event)
                         -- queue up script
                         -- Schedule timer to check when to pop smoke
-                        timer.scheduleFunction(csar.checkWoundedGroupStatus, { _heliName, _woundedName }, timer.getTime() + 5)
+                 --       timer.scheduleFunction(csar.checkWoundedGroupStatus, { _heliName, _woundedName }, timer.getTime() + 5)
```
Why did you remove this? Been a while since I looked but think you need this as an alternative to the f10? maybe add an option to pick between automated and f10?

marcos2221 replied:
you're good..  I didn't remember why I removed it from there..  but it had to be there.  I added the schedule when Insert the unit with prefix in the function below, but if the player gets out and get back to the same unit for sure the will be some issues.  So I added it back in the latest commit.

ciribob commented on CSAR.lua:
```diff
@@ -1597,13 +1646,91 @@ function csar.checkOnboard(_unitName)
 
     timer.scheduleFunction(csar.addMedevacMenuItem, nil, timer.getTime() + 5)
 
+    local _allHeliGroups = coalition.getGroups(coalition.side.BLUE, Group.Category.HELICOPTER)
```
People have done CSAR with non helicopters like the TF-51 :D

marcos2221 replied:
That is for the prefix purpose..   Since I added back the Unit list  i don't think to loop all the planes here as well will be needed.. I added some clarifications that the prefix is meant for Helicopter units.

marcos2221 committed:
Update CSAR.lua

Added back the Fixed units names

-Changed the _lookKeyHeli cause was having issues to get the Id when the unit was nil,  and needed to reset some variables there.. Some messages where not showing up, for example the You're close ..  if you died after the message was shown you never see the message again with the same  unit / downed pilot   csar.helicloseMessage now is reset when the unit dissapears.
Same issue with the Landed status variable I use to make the pilots run..

marcos2221 committed:
Update CSAR.lua

Corrrected Pick up time bug
Added Function csar.spawnCsarAtZone( _zone, _coalition, _description, _randomPoint)

marcos2221 committed:
Create CSAR TEST1.miz

ciribob commented:
Happy with this! Thank you so much. I'll merge now
